### PR TITLE
Add tests of support types of partition values

### PR DESCRIPTION
**Pull request by kneth**
Adding MongoDB Realm apps for each support type of partition value.

The tests should be refactored to avoid code duplication.

### CODE REVIEW

File: tests/js/pv-tests.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        await realm2.syncSession.downloadAllServerChanges();
+
+        let dogs = realm2.objects("Dog");
+        TestCase.assertEqual(dogs.length, 1);
+
+        realm2.close();
+        await user.logOut();
+    },
+
+    /*async testUuidString() {
```
Comment (fronck) [inline tests/js/pv-tests.js:147]: [Question]  Are these commented-out tests waiting for something to be enabled in Stitch, or were they meant to be removed?

**Review by fronck — CHANGES_REQUESTED** [review]
I have a request that we be able to change the local port that is forwarded til Docker.  Since we're changing `test.sh` et. al. anyway.  LMK if it's out of scope.

File: tests/js/pv-tests.js (hunk 1 above)
Comment (steffenagger) [reply tests/js/pv-tests.js:147]: Yes, these were awaiting a core update.

File: scripts/test.sh [hunk 2]
```diff
@@ -66,12 +66,16 @@ start_server() {
...
+      DOCKER_VOLUMES="$DOCKER_VOLUMES -v ${SRCROOT}/tests/mongodb/${app}:/apps/${app}"
+    done
     echo "using object-store stitch dependency: ${MDBREALM_TEST_SERVER_TAG}"
     if [[ -n "$RUN_STITCH_IN_FORGROUND" ]]; then
       # we don't worry about tracking the STITCH_DOCKER_ID because without the -d flag, this docker is tied to the shell
-      docker run -v "${SRCROOT}/tests/mongodb:/apps/os-integration-tests" -p 9090:9090 -it "docker.pkg.github.com/realm/ci/mongodb-realm-test-server:${MDBREALM_TEST_SERVER_TAG}"
+      docker run $DOCKER_VOLUMES -p 9090:9090 -it "docker.pkg.github.com/realm/ci/mongodb-realm-test-server:${MDBREALM_TEST_SERVER_TAG}"
     else
-      STITCH_DOCKER_ID=$(docker run -d $BACKGROUND_FLAG -v "${SRCROOT}/tests/mongodb:/apps/os-integration-tests" -p 9090:9090 -it "docker.pkg.github.com/realm/ci/mongodb-realm-test-server:${MDBREALM_TEST_SERVER_TAG}")
+      STITCH_DOCKER_ID=$(docker run -d $BACKGROUND_FLAG $DOCKER_VOLUMES -p 9090:9090 -it "docker.pkg.github.com/realm/ci/mongodb-realm-test-server:${MDBREALM_TEST_SERVER_TAG}")
```
[the remark this answers is not in the document]
Comment (steffenagger) [reply scripts/test.sh:78]: I support the suggestion, but for testing on iOS/Android the task becomes less trivial, as the port would have to be passed to the test-app & Android need to have the port reversed. So I think out of scope, just now.

[the remark this answers is not in the document]
Comment (fronck) [reply scripts/test.sh:78]: There's other places in the script files that we provide a default if the environment variable isn't set, though.

File: tests/mongodb/pv-int-tests/config.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+    "app_id": "pv-int-tests-lmuyx",
```
[the remark this answers is not in the document]
Comment (steffenagger) [reply tests/mongodb/pv-int-tests/config.json:2]: These ids are overridden on each run (I think we're awaiting an update, where this won't happen).
I think we should see this solution as temporary.

**Comment by kneth** [comment]
Moved to #3677

**Pull request closed without merging**